### PR TITLE
feat: add flag to disable all direct connections instance-wide

### LINKED ISSUE #7422: No way to reliably disable direct connections
Per https://github.com/coder/coder/issues/6791, even when removing all the STUN addresses, there is no strong guarantee that all user <-> workspace traffic will flow through a relay. We should add a flag (maybe `--disable-direct` or `--network-mode=[direct,relay]`) that serves this specific purpose.

### PR DESCRIPTION
**Pull request by deansheather**
Adds `coder server` flag `--disable-direct` which does the following:
- Removes all STUN ports from the DERP map by setting them to `-1` and fully removes any STUNOnly nodes
- Tells agents (on startup) to disable P2P
- Tells clients to disable P2P when connecting to workspaces
- coderd will disable P2P when initiating connections to workspaces i.e. for workspace apps

## TODO:
- [x] Also block for coderd -> workspace direct connections
- [x] Add tests to ensure P2P isn't possible with this flag enabled

Closes #7422

### CODE REVIEW

File: coderd/workspaceagents.go [hunk 1]
```diff
@@ -733,7 +735,8 @@ func (api *API) workspaceAgentConnection(rw http.ResponseWriter, r *http.Request
 	ctx := r.Context()
 
 	httpapi.Write(ctx, rw, http.StatusOK, codersdk.WorkspaceAgentConnectionInfo{
-		DERPMap: api.DERPMap,
+		DERPMap:                api.DERPMap,
+		AllowDirectConnections: !api.DeploymentValues.DERP.Config.DisableDirect.Value(),
```
Comment (kylecarbs) [inline coderd/workspaceagents.go:739]: Can we call this DisableDirectConnections too for consistency?

File: codersdk/deployment.go [hunk 2]
```diff
@@ -710,6 +711,15 @@ when required by your organization's security policy.`,
 			Group:       &deploymentGroupNetworkingDERP,
 			YAML:        "relayURL",
 		},
+		{
+			Name:        "Disable Direct Connections",
+			Description: "Disable peer-to-peer (aka. direct) workspace connections. All workspace connections from the CLI will be proxied through Coder (or custom configured DERP servers) and will never be peer-to-peer when enabled. Workspaces may still reach out to STUN servers to get their address until they are restarted after this change has been made, but new connections will still be proxied regardless.",
+			Flag:        "disable-direct",
```
Comment (kylecarbs) [inline codersdk/deployment.go:717]: I think disable-direct-connections would be a bit easier to understand in isolation.

File: coderd/workspaceagents.go (hunk 1 above)
Comment (coadler) [reply coderd/workspaceagents.go:739]: Agree here. I find it weird we go from `block direct (coderd flag) -> allow direct (coderd api) -> block direct (tailnet)`

**Review by coadler — APPROVED** [review]

File: codersdk/deployment.go (hunk 2 above)
Comment (deansheather) [reply codersdk/deployment.go:717]: `disable-direct-connections` is going to be used as a global flag in the CLI in another PR, so I'm changing this to `block-direct-connections`.

**Review by coadler — APPROVED** [review]

**Pull request merged**